src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,80 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    static final int DEFAULT_CAPACITY_MULTIPLIER = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if ((node.key == null && key == null)
+                    || (node.key != null && node.key.equals(key))) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        table[index] = new Node<>(key, value, table[index]);
+        size++;
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            transfer();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if ((node.key == null && key == null)
+                    || (node.key != null && node.key.equals(key))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void transfer() {
```

Comment:
it's not transfer logic
```suggestion
    private void expand() {
```